nexus/assets/js/home.js

```diff
@@ -0,0 +1,144 @@
+import { connectChat } from "./chat.js"
+import { Socket } from "phoenix";
+
+const pcConfig = { iceServers: [{ urls: "stun:stun.l.google.com:19302" }] };
+const ownVideoPlayer = document.getElementById("videoplayer-own");
+const videoPlayerWrapper = document.getElementById("videoplayer-wrapper");
+
+let localStream = undefined;
+let channel = undefined;
+let pc = undefined;
+let localTracksAdded = false;
+let candidates = [];
+
+async function createPeerConnection() {
+  pc = new RTCPeerConnection(pcConfig);
+
+  pc.ontrack = event => {
+    if (event.track.kind == "video") {
+      console.log("Creating new video element");
+
+      const trackId = event.track.id;
+      const videoPlayer = document.createElement("video");
+      videoPlayer.srcObject = event.streams[0];
+      videoPlayer.autoplay = true;
+      videoPlayer.classList.add("m-auto", "rounded-xl", "max-h-full", "max-w-full");
+
+      videoPlayerWrapper.appendChild(videoPlayer);
+
+      event.track.onended = _ => {
+        console.log("Track ended: " + trackId);
+        videoPlayerWrapper.removeChild(videoPlayer);
+      }
+    } else {
+        // Audio tracks are associated with the stream (`event.streams[0]`) and require no separate actions
+        console.log("New audio track added");
+    }
+  }
+
+  pc.onicegatheringstatechange = () => console.log("Gathering state change: " + pc.iceGatheringState);
+  pc.onconnectionstatechange = () => console.log("Connection state change: " + pc.connectionState);
+  pc.onicecandidate = event => {
+    if (event.candidate == null) {
+      console.log("Gathering candidates complete");
+      forwardSdpAnswer();
```

Comment:
You can push SDP answer immediately after generating it. WebRTC API was designed in a way where you can send every single message as soon as possible. So the flow should be like this:

1. Receive offer
2. Apply offer
3. Create answer
4. Apply answer
5. Send answer to the other side
6. Every time there is a new candidate push it to the other side

Candidates always have to refer to some mline from the SDP offer/answer so they cannot appear before creating local description